internal/node/hybrid/validator.go

```diff
@@ -107,20 +108,66 @@ func validateRolesAnywhereNode(node *api.NodeConfig) error {
 	}
+
+	// IAM roles anywhere certificate validation
 	if node.Spec.Hybrid.IAMRolesAnywhere.CertificatePath == "" {
 		return fmt.Errorf("CertificatePath is missing in hybrid iam roles anywhere configuration")
 	}
-	if node.Spec.Hybrid.IAMRolesAnywhere.PrivateKeyPath == "" {
-		return fmt.Errorf("PrivateKeyPath is missing in hybrid iam roles anywhere configuration")
-	}
-
 	if !file.Exists(node.Spec.Hybrid.IAMRolesAnywhere.CertificatePath) {
 		return fmt.Errorf("IAM Roles Anywhere certificate %s not found", node.Spec.Hybrid.IAMRolesAnywhere.CertificatePath)
 	}
+	if err := ValidateCertificate(node.Spec.Hybrid.IAMRolesAnywhere.CertificatePath, nil); err != nil {
+		var validationErr *ValidationError
+		if errors.As(err, &validationErr) {
+			updateIAMRAErrorMessage(node.Spec.Hybrid.IAMRolesAnywhere.CertificatePath, validationErr)
+		}
+		return fmt.Errorf("IAM Roles Anywhere certificate error - %v", err)
+	}
 
+	// IAM roles anywhere key validation
+	if node.Spec.Hybrid.IAMRolesAnywhere.PrivateKeyPath == "" {
+		return fmt.Errorf("PrivateKeyPath is missing in hybrid iam roles anywhere configuration")
+	}
 	if !file.Exists(node.Spec.Hybrid.IAMRolesAnywhere.PrivateKeyPath) {
 		return fmt.Errorf("IAM Roles Anywhere private key %s not found", node.Spec.Hybrid.IAMRolesAnywhere.PrivateKeyPath)
 	}
 
 	return nil
 }
+
+// updateIAMRAErrorMessage update the IAM Roles Anywhere specific remediation messages
+func updateIAMRAErrorMessage(certPath string, err *ValidationError) {
+	if err == nil {
+		return
+	}
+
+	switch err.validationErrorType {
+	case ErrorNoCert:
+		err.message = "no iam-roles-anywhere certificate found"
+		err.remediation = fmt.Sprintf("Certificate file not found at %s. Please provide a valid IAM Roles Anywhere certificate.", certPath)
+	case ErrorCertFile:
+		err.message = "checking iam-roles-anywhere certificate"
+		err.remediation = fmt.Sprintf("Certificate file not found at %s. Please provide a valid IAM Roles Anywhere certificate.", certPath)
+	case ErrorReadFile:
+		err.message = "reading iam-roles-anywhere certificate"
+		err.remediation = fmt.Sprintf("Certificate file not found at %s. Please provide a valid IAM Roles Anywhere certificate.", certPath)
+	case ErrorInvalidFormat:
+		err.message = "parsing iam-roles-anywhere certificate"
+		err.remediation = fmt.Sprintf("The IAM Roles Anywhere certificate at %s is not in a valid format. Please provide a valid certificate.", certPath)
+	case ErrorClockSkewDetected:
+		err.message = "iam-roles-anywhere certificate is not yet valid"
+		err.remediation = fmt.Sprintf("Verify the system time. Current system time is behind IAM Roles Anywhere certificate's start time at %s", certPath)
+	case ErrorExpired:
+		err.message = "iam-roles-anywhere certificate has expired"
+		err.remediation = fmt.Sprintf("IAM Roles Anywhere certificate at %s has expired. Please provide a valid, non-expired certificate.", certPath)
+	case ErrorParseCA:
```

Comment:
is this even possible?

Reply:
I was talking about the `ErrorParseCA`. afaik we don't have the ca that signs the iam-ra cert?